docs/resources/source_mysql.md

```diff
@@ -0,0 +1,111 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "materialize_source_mysql Resource - terraform-provider-materialize"
+subcategory: ""
+description: |-
+  A MySQL source describes a MySQL instance you want Materialize to read data from.
+---
+
+# materialize_source_mysql (Resource)
+
+A MySQL source describes a MySQL instance you want Materialize to read data from.
+
+## Example Usage
+
+```terraform
+resource "materialize_source_mysql" "test" {
+  name          = "source_mysql"
+  schema_name   = materialize_schema.test.name
+  database_name = materialize_database.test.name
+
+  cluster_name = "quickstart"
+
+  mysql_connection {
+    name = materialize_connection_mysql.test.name
+  }
+
+  table {
+    name  = "shop.mysql_table1"
+    alias = "alias_mysql_table1"
+  }
+  table {
+    name  = "shop.mysql_table2"
+    alias = "alias_mysql_table2"
+  }
+}
+
+# CREATE SOURCE schema.source_mysql
+#   FROM MYSQL CONNECTION "database"."schema"."pg_connection" (PUBLICATION 'mz_source')
+#   FOR TABLES (schema1.table_1 AS s1_table_1, schema2_table_1 AS s2_table_1);
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `mysql_connection` (Block List, Min: 1, Max: 1) The MySQL connection to use in the source. (see [below for nested schema](#nestedblock--mysql_connection))
+- `name` (String) The identifier for the source.
+
+### Optional
+
+- `cluster_name` (String) The cluster to maintain this source.
+- `comment` (String) **Public Preview** Comment on an object in the database.
+- `database_name` (String) The identifier for the source database. Defaults to `MZ_DATABASE` environment variable if set or `materialize` if environment variable is not set.
+- `ownership_role` (String) The owernship role of the object.
+- `region` (String) The region to use for the resource connection. If not set, the default region is used.
+- `schema_name` (String) The identifier for the source schema. Defaults to `public`.
```

Comment:
this is slightly confusing since it refers to the schema on the materialize side that holds the source object, not the schema of the upstream mysql database where the tables are. Is there a way to reword this to be more clear?

Reply:
Just rephrased this! Let me know what you think.

Reply:
sounds better, thanks!